src/AuthProvider.tsx

```diff
@@ -134,6 +141,14 @@ const unsupportedEnvironment = (fnName: string) => () => {
 const defaultUserManagerImpl =
     typeof window === "undefined" ? null : UserManager;
 
+type NavigatorArgTypes =
```

Comment:
I guess this is not really needed, i would like to keep the code below using `...args: never[]`...

Reply:
Thank you for your review and comment.

Can not keep `...args: never[]`, because `ProcessResourceOwnerPasswordCredentialsArgs`'s `username` and `password` are required.

I considered three modification patterns.

May I fix the code to "Pattern 1 : inline type definition"?

<details>
<summary>Pattern 1 : inline type definition</summary>

1. Remove NavigatorArgTypes.
2. args type to `ProcessResourceOwnerPasswordCredentialsArgs | never`

```ts
                           ? async (args: ProcessResourceOwnerPasswordCredentialsArgs | never) => {
                                dispatch({
                                    type: "NAVIGATOR_INIT",
                                    method: key,
                                });
                                try {
                                    return await userManager[key](args);
```

</details>


<details>
<summary>Pattern 2 : fix NavigatorArgTypes</summary>

1. Modify NavigatorArgTypes to `ProcessResourceOwnerPasswordCredentialsArgs | never`

```ts
type NavigatorArgTypes = ProcessResourceOwnerPasswordCredentialsArgs | never;
```

2. args type keep `NavigatorArgTypes`

```ts
                           ? async (args: NavigatorArgTypes) => {
                                dispatch({
                                    type: "NAVIGATOR_INIT",
                                    method: key,
                                });
                                try {
                                    return await userManager[key](args);
```

</details>

Pattern 3 : now PR code keep.

Reply:
ah i see, the others were optional...

Maybe pattern 1 is not that invasive but still readable. I shortly tried but the pattern 1 gives me a type error on `return await userManager[key](args);`, have not invested much time yet...

Reply:
>  I shortly tried but the pattern 1 gives me a type error on `return await userManager[key](args);`

Sorry, I made a mistake type definition.
The correct type is `ProcessResourceOwnerPasswordCredentialsArgs & never[]`. (`&` instead of `|`)

Reply:
makes sense